src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagCache.cs

```diff
@@ -0,0 +1,64 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Core.Features.CustomTag
+{
+    public class CustomTagCache : ICustomTagCache
+    {
+        private CustomTagList _customtags;
+        private ICustomTagStore _customTagStore;
+        private TimeSpan _expirationInterval;
+        private DateTimeOffset? _lastRefreshTime;
+
+        public CustomTagCache(ICustomTagStore customTagStore, TimeSpan expirationInterval)
```

Comment:
Why is the expiration passed in?

Reply:
The value will be passed from config file.
I prefer an centralized place for ExpirationInterval and CustomTagPropagationTime for easy management.